### PR TITLE
Add new examples and update existing ones

### PR DESCRIPTION
**Pull request by RDIL**

### CODE REVIEW

File: .spelling [hunk 1]
```diff
@@ -1,4 +1,5 @@
 256-bit
```
Comment (RDIL) [inline .spelling]: noticed I screwed up a few prs back so I fixed the order

Comment (fkorotkov) [reply .spelling]: I did sort it back here https://github.com/cirruslabs/cirrus-ci-docs/commit/d56a6de9298470420c5215b0ef27e0b27b4352fd#diff-b9a2754b5f195f0113dc72bd305fcdb3

Comment (RDIL) [reply .spelling]: Strange - lots of things are out of order if you look at this PRs diff

**Comment by fkorotkov** [comment]
I don't think we need a new "Java and Kotlin" section. If you want to add a maven example, you can simply create a new section (note that examples are sorted alphabetically). What do you think?

**Comment by RDIL** [comment]
Good plan.

File: docs/examples.md [hunk 2]
```diff
@@ -41,6 +41,18 @@ check_android_task:
...
     Please don't forget to setup [Remote Build Cache](#build-cache) for your Gradle project. Or at least [simple folder caching](#gradle-caching).
 
+## Apache Maven
+
+Here is an example `.cirrus.yml` that builds a Maven project (and runs basic tests on code to make sure there are no simple errors):
+```yaml
+task:
+  name: Cirrus CI
+  container:
+    image: maven:3-jdk-8
```
Comment (fkorotkov) [inline docs/examples.md]: JDK 11 is actually the default one. You can probably change the tag to `maven:3` which is `maven:3-jdk11`.

File: docs/examples.md [hunk 3]
```diff
@@ -41,6 +41,18 @@ check_android_task:
 !!! info
     Please don't forget to setup [Remote Build Cache](#build-cache) for your Gradle project. Or at least [simple folder caching](#gradle-caching).
 
+## Apache Maven
+
+Here is an example `.cirrus.yml` that builds a Maven project (and runs basic tests on code to make sure there are no simple errors):
+```yaml
```
Comment (fkorotkov) [inline docs/examples.md]: Add a new line before.

File: docs/examples.md [hunk 4]
```diff
@@ -41,6 +41,18 @@ check_android_task:
 !!! info
     Please don't forget to setup [Remote Build Cache](#build-cache) for your Gradle project. Or at least [simple folder caching](#gradle-caching).
 
+## Apache Maven
+
+Here is an example `.cirrus.yml` that builds a Maven project (and runs basic tests on code to make sure there are no simple errors):
```
Comment (fkorotkov) [inline docs/examples.md]: You can probably use similar working like this:

```suggestion
Official [Maven Docker images](https://hub.docker.com/_/maven/) can be used for building and testing Maven projects:
```

here to be more consistent.

File: docs/examples.md [hunk 5]
```diff
@@ -41,6 +41,18 @@ check_android_task:
 !!! info
     Please don't forget to setup [Remote Build Cache](#build-cache) for your Gradle project. Or at least [simple folder caching](#gradle-caching).
 
+## Apache Maven
```
Comment (fkorotkov) [inline docs/examples.md]: I don't think many people use the full name. You can simply use *Maven*.

File: docs/examples.md [hunk 6]
```diff
@@ -41,6 +41,18 @@ check_android_task:
...
+
+Here is an example `.cirrus.yml` that builds a Maven project (and runs basic tests on code to make sure there are no simple errors):
+```yaml
+task:
+  name: Cirrus CI
+  container:
+    image: maven:3-jdk-8
+test_script: mvn compile -B
+```
+If you want to use a different JDK version, you can change the `jdk-8` part to whatever version you want (e.g. `jdk-11`).
```
Comment (fkorotkov) [inline docs/examples.md]: Don't think there is a need for it. Docker Hub page has all the information.

File: docs/examples.md [hunk 7]
```diff
@@ -197,6 +196,33 @@ org.gradle.configureondemand=true
 org.gradle.jvmargs=-Dfile.encoding=UTF-8
 ```
 
+## JUnit
+
+Here is a `.cirrus.yml` that (once completed), runs and uploads JUnit results:
```
Comment (fkorotkov) [inline docs/examples.md]:
```suggestion
Here is a `.cirrus.yml` that (once succeeded or failed), parses and uploads JUnit reports:
```

File: docs/examples.md [hunk 8]
```diff
@@ -197,6 +196,33 @@ org.gradle.configureondemand=true
...
+
+## Maven
+
+Official [Maven Docker images](https://hub.docker.com/_/maven/) can be used for building and testing Maven projects:
+
+```yaml
+task:
+  name: Cirrus CI
+  container:
+    image: maven:3-jdk-11
```
Comment (fkorotkov) [inline docs/examples.md]:
```suggestion
    image: maven:latest
```

File: docs/examples.md [hunk 9]
```diff
@@ -197,6 +196,33 @@ org.gradle.configureondemand=true
...
+test_script: mvn compile -B
```
Comment (fkorotkov) [inline docs/examples.md]:
```suggestion
test_script: mvn test -B
```

File: .spelling (hunk 1 above)
Comment (fkorotkov) [reply .spelling]: Are you sure these changes are needed? I don't see much new words except maybe *Maven* and *Junit*. Why there are so many changes when master is green at the moment.

File: docs/examples.md (hunk 9 above)
Comment (RDIL) [reply docs/examples.md]: This doesn't work unless tests are specified

Comment (fkorotkov) [reply docs/examples.md]: Are you sure? Back in the days it definitely worked. 🤔

Comment (RDIL) [reply docs/examples.md]: Just earlier today I set up a maven project, and running test returned no tests specified (but still exited with 0).  Compile checks for basic syntax errors.

Comment (fkorotkov) [reply docs/examples.md]: Do you have any tests though?

Comment (RDIL) [reply docs/examples.md]: None set up.

Comment (fkorotkov) [reply docs/examples.md]: That's maybe the issues. If you'll have any tests (which you should), `mvn test` will work.

File: docs/examples.md [hunk 10]
```diff
@@ -197,6 +196,36 @@ org.gradle.configureondemand=true
...
+Official [Maven Docker images](https://hub.docker.com/_/maven/) can be used for building and testing Maven projects:
+
+```yaml
+task:
+  name: Cirrus CI
+  container:
+    image: maven:latest
+  maven_cache:
+    folder: ~/.m2
+    fingerprint_script: cat pom.xml
```
Comment (fkorotkov) [inline docs/examples.md]: It won't work for a project with multiple modules. I think you can simply remove `fingerprint_script`.

**Comment by fkorotkov** [comment]
Thank you!

**Pull request merged**